1 Million limit on lettuceRediSearchClient.findall

this works at 1000000 but fails at 10000001
 Set<ProductEntity> products = lettuceRediSearchClient.findAll(1000001).resultStream()

com.rnbwarden.redisearch.LettuceTest > testFindAll FAILED
    io.lettuce.core.RedisCommandExecutionException at LettuceTest.java:184
        Caused by: io.lettuce.core.RedisCommandExecutionException at ExceptionFactory.java:135

this is the code link:
https://github.com/rnbWarden/jredisearch-spring-boot-starter/blob/master/src/test/java/com/rnbwarden/redisearch/LettuceTest.java

--- a/src/aggregate/aggregate_request.c
+++ b/src/aggregate/aggregate_request.c
@@ -135,7 +135,7 @@ static int handleCommonArgs(AREQ *req, ArgsCursor *ac, QueryError *status, int a
     if (arng->limit == 0) {
       // LIMIT 0 0
       req->reqflags |= QEXEC_F_NOROWS;
-    } else if (arng->limit > SEARCH_REQUEST_RESULTS_MAX) {
+    } else if ((arng->limit > SEARCH_REQUEST_RESULTS_MAX) && (req->reqflags & QEXEC_F_IS_SEARCH)) {
       QueryError_SetErrorFmt(status, QUERY_ELIMIT, "LIMIT exceeds maximum of %llu",
                              SEARCH_REQUEST_RESULTS_MAX);
       return ARG_ERROR;

--- a/src/pytest/test_aggregate.py
+++ b/src/pytest/test_aggregate.py
@@ -395,6 +395,10 @@ class TestAggregate():
                           'LIMIT', 0, 20000000)
         self.env.assertEqual(2266, len(rv))
 
+        # SEARCH should fail
+        self.env.expect('ft.search', 'games', '*', 'limit', 0, 2000000).error()     \
+                .contains('LIMIT exceeds maximum of 1000000')
+        
     # def testLoadAfterSortBy(self):
     #     with self.env.assertResponseError():
     #         self.env.cmd('ft.aggregate', 'games', '*',

--- a/src/pytest/test_aggregate.py
+++ b/src/pytest/test_aggregate.py
@@ -390,6 +390,11 @@ class TestAggregate():
                           'SORTBY', 2, '@brand', 'ASC')
         self.env.assertEqual('__generated_aliasfirst_valuetitle,by,price,desc', rv[1][2])
 
+    def testIssue1125(self):
+        rv = self.env.cmd('ft.aggregate', 'games', '*',
+                          'LIMIT', 0, 20000000)
+        self.env.assertEqual(2266, len(rv))
+
     # def testLoadAfterSortBy(self):
     #     with self.env.assertResponseError():
     #         self.env.cmd('ft.aggregate', 'games', '*',

